### ISSUE TITLE
Can't use default extras (dicts)

### ISSUE THREAD

**Issue by eranzim** [googleprojectzero/winafl#354]
When I try to use the default extra files (e.g. gif.dict from testcases\_extras folder) with the -x flag with afl-fuzz.exe, I get this error:
```
[-] PROGRAM ABORT : Extra '...path\to\gif.dict' is too big (298 B, limit is 128 B)
         Location : load_extras(), ...path\to\winafl\afl-fuzz.c:1931
```
It seems like the default gif.dict doesn't work - maybe it should be minimized or the threshold should be increased?

**Comment by ifratric** [comment]
Could you give me a command line to reproduce this? There are two ways the -x flag can be used: 
 - you could point it to an individual file which contains all the tokens
 - you could point it to a directory where each file in the directory is considered a token
so I'm wondering how you used it. In this case gif.dict is not a token, but rather a file containing the tokens and should be parsed as such.

afl-fuzz _should_ detect if you pointed it to a file or directory, see https://github.com/googleprojectzero/winafl/blob/55bde9dad30854e33e625cf6565d450bac8e1472/afl-fuzz.c#L1920. But perhaps that check is wrong and thinks it's directory even though you pointed it to the exact file?

**Comment by ifratric** [comment]
If this check is failing, then we could do a better check with GetFileAttributesA

**Comment by eranzim** [comment]
So maybe I misused it? I pointed it at a directory (I think pointing it at a file failed, but I'll have to recheck) containing several dict files (gif.dict, jpeg.dict, etc.).
If, for example, I want to use all image-related dict files, I need to concatenate them all to a single file (simple file concatenation)?

**Comment by ifratric** [comment]
Right, either (a) concatenate it all in a single file, or (b) extract all tokens from all files as separate files to a directory. (a) is probably simpler.
